base_station/gui/src/components/SASiteControls.vue

```diff
@@ -89,11 +91,155 @@
           'type': 'MicroCam',
           'id': 'camera_' + this.site
         })
+      },
+      swapComponent: function(component)
+      {
+        this.currentComponent = component;
       }
     },
 
     components: {
-      Checkbox
+      Checkbox,
+      'Chlorophyll': {
+        template: '<div><p>Turn off White LEDs</p><button id="execute" @click="swapComponent(\'WhiteLEDsOff\')">Execute</button></div>',
+        methods: {
+          swapComponent: function(component)
+          {
+            this.$parent.$parent.$parent.publish("/mosfet_cmd", {
+              'type': 'MosfetCmd',
+              'device': 2,
+              'enable': false
+            })
+            var site_letter = 'b';
+            if (this.$parent.site == "White") {
+              site_letter = 'w';
+            }
+            else if (this.$parent.site == "Yellow") {
+              site_letter = 'y';
+            }
+            // alert(site_letter);
+            this.$parent.$parent.$parent.publish("/spectral_cmd", {
+              'type': 'SpectralCmd',
+              'sensor': site_letter
+            })
+            this.$emit("click", this.$parent.swapComponent(component))
+          }
+        }
+      },
+      'WhiteLEDsOff': {
+        template: '<div><p>Turn on UV LED</p><button id="execute" @click="swapComponent(\'UVLEDOn\')">Execute</button></div>',
+        methods: {
+          swapComponent: function(component)
+          {
+            this.$parent.$parent.$parent.publish("/mosfet_cmd", {
+              'type': 'MosfetCmd',
+              'device': 1,
```

Comment:
instead of numbers can you use the constants in the struct?

Reply:
This should be resolved